### PR TITLE
add macos runner

### PR DESCRIPTION
**Pull request by umireon**

### CODE REVIEW

**Comment by umireon** [comment]
@zkochan Review ready.

**Review by zkochan — APPROVED** [review]

File: .github/workflows/install-macos.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          - install.sh
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install shell
+        run: brew install ${{ matrix.shell }}
+        if: ${{ matrix.shell == 'ksh' || matrix.shell == 'dash' }}
+      - name: Run installation script
+        run: SHELL="$(which ${{ matrix.shell }})" ${{ matrix.shell }} ${{ matrix.script }}
```
Comment (zkochan) [inline .github/workflows/install-macos.yml:29]: Why should the SHELL env variable be set?

Comment (umireon) [reply .github/workflows/install-macos.yml:29]: I recognize that `pnpm setup` refers SHELL env to configure the shell profile.
I want to check in GitHub Actions whether this works.
Running `${{ matrix.shell }} ${{ matrix.script }}` will leave the SHELL env the system's default.
I tried `-l` option but the SHELL env won't change.

Comment (zkochan) [reply .github/workflows/install-macos.yml:29]: Hm, maybe we should use a different way to recognize the active SHELL. Or maybe it is OK to use the default SHELL.

Comment (umireon) [reply .github/workflows/install-macos.yml:29]: I think CI needs SHELL to be changed for a testing purpose but a user doesn't need to be changed.
As a user, `pnpm setup` should recognize the login shell, not the running shell.
The login shell and the script running shell often differ.

**Pull request merged**